IGC/OCLFE/igd_fcl_mcl/source/clang_tb.cpp

```diff
@@ -771,34 +771,49 @@ namespace TC
 
         const char * optValue = optSubstring + optName.size();
         const char * end = optValue + strlen(optValue);
+        std::string_view opt(optValue, end - optValue);
 
         // parse
-        const std::string invalidFormatMessage = "Invalid format of -cl-std option, expected -cl-std=CLMAJOR.MINOR";
-        auto isNumeric = [](char v) { return (v >= '0') && (v <= '9'); };
-        if (false == ((end - optValue >= 5) && (optValue[0] == 'C') && (optValue[1] == 'L') && isNumeric(optValue[2])
-            && (optValue[3] == '.') && isNumeric(optValue[4])
-            )
-            ) {
-            exceptString = invalidFormatMessage;
-            return 0;
-        }
-
         unsigned int retVersion = 0;
-        // subverions
-        if ((end - optValue >= 7) && (optValue[5] != ' ')) {
-            if ((optValue[5] == '.') || isNumeric(optValue[6])) {
-                retVersion += optValue[6] - '0';
+        if (opt.find("CLC++") == 0) {
+            if (opt == "CLC++" || opt == "CLC++1.0") {
+                retVersion = 200;
             }
-            else if (isNumeric(optValue[5])) {
-                retVersion += optValue[5] - '0';
+            else if (opt == "CLC++2021") {
+                retVersion = 300;
             }
             else {
+                const std::string invalidFormatMessage = "Invalid format of -cl-std option, expected -cl-std=CLC++, -cl-std=CLC++1.0, or -cl-std=CLC++2021";
```

Comment:
and here we need to assign `invalidFormatMessage` to the `exceptString` and return 0. 
```suggestion
                exceptString = invalidFormatMessage;
                return 0;
```